src/frontend/src/optimizer/plan_node/stream_group_topn.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use risingwave_common::util::sort_util::OrderType;
+
+use super::utils::TableCatalogBuilder;
+use super::PlanBase;
+use crate::optimizer::property::{Distribution, Order};
+use crate::{PlanRef, TableCatalog};
+
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
+pub struct StreamGroupTopN {
+    pub base: PlanBase,
+    group_key: Vec<usize>,
+    limit: usize,
+    offset: usize,
+    order: Order,
+}
+
+#[allow(dead_code)]
+impl StreamGroupTopN {
+    pub fn new(
+        input: PlanRef,
+        group_key: Vec<usize>,
+        limit: usize,
+        offset: usize,
+        order: Order,
+    ) -> Self {
+        let dist = match input.distribution() {
+            Distribution::Single => Distribution::Single,
+            _ => panic!(),
+        };
+
+        let base = PlanBase::new_stream(
+            input.ctx(),
+            input.schema().clone(),
```

Comment:
group key might be moved to the first few columns? so schema might need rewriting (based on how you implemented the executor, I forget about whether you put the group by columns to the front or not)

Reply:
The columns specified by the group key does not appear to be added to the front of the schema in the operator and the sechama seems to be stored as it is. I only want to specified the order of the states in the state table.

Reply:
the group TopN is different with groupAgg because all columns in input will appear in the output